File: LC226.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        
        if(not root)
            return NULL;
        if(root->left and root->right){
            TreeNode *t = root->right;
            root->right = invertTree(root->left);
            root->left = invertTree(t);
        }
        else if(root->left and not root->right){
            root->right = invertTree(root->left);
            root->left = NULL;
        }
        else if(root->right and not root->left){
            root->left = invertTree(root->right);
            root->right = NULL;
        }
        else 
            return root;
    
        return root;
    }
};

File: LC226.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:

        if not root:
            return None
        
        if root.left and root.right:
            t = root.right
            root.right = self.invertTree(root.left)
            root.left = self.invertTree(t)
        elif root.left and not root.right:
            root.right = self.invertTree(root.left)
            root.left = None
        elif root.right and not root.left:
            root.left = self.invertTree(root.right)
            root.right = None
        else:
            return root
        return root